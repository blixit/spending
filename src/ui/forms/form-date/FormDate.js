import React from 'react';
import styled from 'styled-components';

import DatePickerUI from 'ui/inputs/date-picker/DatePicker';
import LabelUI from 'ui/forms/form-label/FormLabel';

const ErrorText = styled.div`
  color: red;
`;

const Label = styled(LabelUI)`
  .react-datepicker-wrapper,
  .react-datepicker__input-container,
  .react-datepicker__input-container input {
    text-align: ${({ arabic }) => arabic ? 'right' : 'left'};
    height: 20px;
    lineheight: 18px;
  }
`;

const Container = styled.div`
  padding-top: 5px;
  padding-bottom: 5px;
`;

const FormDate = ({ label, error, selected, type, onChange, arabic, ...rest }) => {
  const dateProps = { selected, type, onChange, arabic, ...rest };
  return (
    <Container {...rest} >
      <Label arabic={arabic}>
        <span>{label}</span>
        <DatePickerUI {...dateProps} />
      </Label>
      {error && <ErrorText>{error}</ErrorText>}
    </Container>
  )
};

FormDate.displayName = 'FormDate';

export default FormDate;
