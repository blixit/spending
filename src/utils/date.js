/**
 * Formats a date for the backend
 * @param {Date} date the date to convert
 */
export const toBackendDate = (date) => {
  return date
    ? date.toLocaleDateString() + ' ' + date.toLocaleTimeString()
    : date;
};