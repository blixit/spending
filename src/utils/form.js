/**
* Removes the listed fields from the source
* @param {*} form The source object
* @param {*} fields The fields list
*/
export const clean = (form, fields) => {
  const result = {};
  // eslint-disable-next-line no-unused-vars
  for (const field in form) {
    if (!form.hasOwnProperty(field) || fields.includes(field)) {
      continue;
    }
    result[field] = form[field];
  }
  return result;
};
