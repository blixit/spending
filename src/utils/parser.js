export const stringify = data => {
  try {
    return JSON.stringify(data);
  } catch (error) {
    return null;
  };
}

export const parse = data => JSON.parse(data);