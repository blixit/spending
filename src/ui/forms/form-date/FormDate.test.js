import render from 'testing/render';

import { withTheme } from 'ui/theming/test.utils';

import FormDate from './FormDate';

describe('[forms/form-date] FormDate', () => {
  it('renders properly', () => {
    const props = {
      label: 'Label',
      type: 'text',
      onChange: jest.fn(),
      placeholderText: 'select the date',
      selected: Date.UTC(2019, 1, 17)
    };
    const tree = render(withTheme(FormDate, props));
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
