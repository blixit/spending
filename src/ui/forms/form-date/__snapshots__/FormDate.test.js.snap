// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[forms/form-date] FormDate renders properly 1`] = `
<div
  className="sc-ifAKCX bQstZT"
>
  <label
    className="sc-bwzfXH cJnpOh sc-bxivhb jqRqDB"
  >
    <span>
      Label
    </span>
    <div
      className="react-datepicker-wrapper"
    >
      <div
        className="react-datepicker__input-container"
      >
        <input
          className="sc-bdVaJa fEZIYW"
          disabled={false}
          onBlur={[Function]}
          onChange={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          placeholder="select the date"
          readOnly={false}
          type="text"
          value="17/02/2019"
        />
      </div>
    </div>
  </label>
</div>
`;
